Source/Peripheral.swift

```diff
@@ -123,7 +123,7 @@ public class Peripheral {
      Immediately after that `.Complete` is emitted.
      */
     public func discoverServices(_ serviceUUIDs: [CBUUID]?) -> Observable<[Service]> {
-        if let identifiers = serviceUUIDs, let services = self.services?.filter({ identifiers.contains($0.uuid) }), identifiers.count == services.count {
+        if let identifiers = serviceUUIDs, let services = self.services?.filter({ identifiers.contains($0.uuid) }), identifiers.count <= services.count {
```

Comment:
yes, I was only modifying the existing code == for counts. I didn't want to do more complex refactoring. I've tested my use case an it worked for that but you are right we need more generic solution. I think the basic principle could be is that whenever we request a list of [A,B] the check should examine if this list is subset of the cached one. If yes then we can safely return the filtered list